src/notation/view/widgets/editstyle.cpp

```diff
@@ -1587,16 +1587,23 @@ void EditStyle::setValues()
     disableVerticalSpread->setChecked(!styleValue(StyleId::enableVerticalSpread).toBool());
 }
 
+mu::io::path EditStyle::selectStyleFile(std::shared_ptr<mu::framework::IInteractive> interactive,
+                                        std::shared_ptr<INotationConfiguration> configuration, bool forLoad)
```

Comment:
I'd rather put this somewhere else but I'm not sure where...

Reply:
I think we need to create a separate interface for user interaction with styles. For example, like this: 

<img width="858" alt="Screenshot 2021-08-18 at 11 08 05" src="https://user-images.githubusercontent.com/67867444/129871179-7a0fdd8b-c188-4d68-ac99-25a38ceb0423.png">

Reply:
After that, we need to move all code from the screenshot to StyleScenario. Because NotationInteraction isn't a proper place for it

<img width="797" alt="Screenshot 2021-08-18 at 09 59 17" src="https://user-images.githubusercontent.com/67867444/129871544-9c537a92-e92d-45f5-a740-98733e8791ce.png">

Reply:
I'm happy to move it to NotationStyle, though NotationInteraction seemed logical enough based on what else it does.

Reply:
Move to NotationStyle, seemed to make sense.

Reply:
I believe, NotationStyle isn't a proper place for the interaction logic (such as asking questions, showing errors, etc). We need to create a separate class for this (I suggested IStyleScenario, but you can use a different name for it). After that, NotationStyle::loadStyle will look like:

```
loadStyle(path) {
    m_undoStack->prepareChanges();
    bool ok = score()->loadStyle(path);

    if (ok) {
        m_undoStack->commitChanges();
    } else {
        m_undoStack->rollbackChanged();
    }

     return ok;
} 
```

And StyleScenarion::loadStyle will look like:

```
loadStyle(notation, path) {
    bool ok = notation->style()->loadStyle(path)
    if (!ok) {
        interactive()->error(...)
    }
...
} 
```

Reply:
That's why I originally put it in NotationInteraction!
I'm not comfortable introducing a whole new class just for these two commands, and "Scenario" seems like the wrong word anyway. Feel free to hand it over to one of the core team if you'd prefer a different implementation, I only picked this because it seemed pretty unlikely to impact anyone else.

Reply:
This is not a good place for this code.
We adhere to the MVVM architectural pattern
https://en.wikipedia.org/wiki/Model%E2%80%93view%E2%80%93viewmodel
User interaction must be at the ViewModel level
This place is Model, there should be no user interaction, ideally. Otherwise, this code (loading, saving styles) cannot be used in other cases, for example, during auto testing, or on the backend, or anywhere else.

Reply:
ViewModel classes are in modulename/view

Reply:
Which is where NotationViewController is, and where I originally had the code to show the dialogs.  I then put the code to do the actual style load/save in NotationInteraction, but Roman didn't approve.  That part's mostly just plumbing anyway, the code that actually does the real work is all inside the engraving library anyway.

Reply:
> Which is where NotationViewController is, and where I originally had the code to show the dialogs. I then put the code to do the actual style load/save in NotationInteraction, but Roman didn't approve. That part's mostly just plumbing anyway, the code that actually does the real work is all inside the engraving library anyway.

Originally, your code for showing dialogs was in NotationInteraction (see my screenshot above) :) NotationInteraction isn't part of view models. So I suggested, that you can create a separate class and reuse it in NotationActionsController and in the EditStyle dialog

Reply:
That's why I don't like squashing commits, I'm sure there was an earlier version I had that put showing the dialogs into NotationActionController but there's no trace of it now.

Reply:
Actually you can see how it was originally here: https://github.com/musescore/MuseScore/pull/8868/commits/6f72b43180bd5c75a1909bfb28d7e2328b8358a4

I didn't really like that either (putting the static helper function in EditStyle), but it did at least adhere to the MVVM pattern.

Reply:
Because you didn't swuash nor rebase yet...

Reply:
Rare occasion of my common sense kicking in then :-P